bpa-services/src/main/java/org/egov/bpa/service/NocService.java

```diff
@@ -0,0 +1,259 @@
+		try {
+			log.debug("Creating NOC application with nocType : " + nocRequest.getNoc().getNocType());
+			responseMap = (LinkedHashMap<String, Object>) serviceRequestRepository.fetchResult(uri, nocRequest);
+			NocResponse nocResponse = mapper.convertValue(responseMap, NocResponse.class);
+			log.debug("NOC created with applicationNo : " + nocResponse.getNoc().get(0).getApplicationNo());
+		} catch (Exception se) {
+			throw new CustomException(BPAErrorConstants.NOC_SERVICE_EXCEPTION,
+					" Failed to create NOC of Type " + nocRequest.getNoc().getNocType());
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void updateNoc(NocRequest nocRequest) {
+		StringBuilder uri = new StringBuilder(config.getNocServiceHost());
+		uri.append(config.getNocUpdateEndpoint());
+
+		LinkedHashMap<String, Object> responseMap = null;
+		try {
+			responseMap = (LinkedHashMap<String, Object>) serviceRequestRepository.fetchResult(uri, nocRequest);
+			NocResponse nocResponse = mapper.convertValue(responseMap, NocResponse.class);
+			log.debug("NOC updated with applicationNo : " + nocResponse.getNoc().get(0).getApplicationNo());
+		} catch (Exception se) {
+			throw new CustomException(BPAErrorConstants.NOC_SERVICE_EXCEPTION,
+					" Failed to update NOC of Type " + nocRequest.getNoc().getNocType());
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<Noc> fetchNocRecords(BPARequest bpaRequest) {
+
+		StringBuilder url = getNOCWithSourceRef(bpaRequest);
+
+		RequestInfoWrapper requestInfoWrapper = RequestInfoWrapper.builder().requestInfo(bpaRequest.getRequestInfo())
+				.build();
+		LinkedHashMap<String, Object> responseMap = null;
+		try {
+			responseMap = (LinkedHashMap<String, Object>) serviceRequestRepository.fetchResult(url, requestInfoWrapper);
+			NocResponse nocResponse = mapper.convertValue(responseMap, NocResponse.class);
+			return nocResponse.getNoc();
+		} catch (Exception e) {
+			throw new CustomException(BPAErrorConstants.NOC_SERVICE_EXCEPTION, " Unable to fetch the NOC records");
+		}
+	}
+
+	/**
+	 * fetch the noc records with sourceRefId
+	 * @param bpaRequest
+	 * @return
+	 */
+	private StringBuilder getNOCWithSourceRef(BPARequest bpaRequest) {
+		StringBuilder uri = new StringBuilder(config.getNocServiceHost());
+		uri.append(config.getNocSearchEndpoint());
+		uri.append("?tenantId=");
+		uri.append(bpaRequest.getBPA().getTenantId());
+		NocRequest nocRequest = new NocRequest();
+		nocRequest.setRequestInfo(bpaRequest.getRequestInfo());
+		uri.append("&sourceRefId=");
+		uri.append(bpaRequest.getBPA().getApplicationNo());
+		return uri;
+	}
+
+	/**
+	 * Calls the iniate  workflow for the applicable noc records
+	 * @param bpaRequest
+	 * @param mdmsData
+	 */
+	public void initiateNocWorkflow(BPARequest bpaRequest, Object mdmsData) {
+		log.debug("====> initiateNocWorkflow");
+		List<Noc> nocs = fetchNocRecords(bpaRequest);
+		log.debug("====> initiateNocWorkflow = no of noc "+ nocs.size());
```

Comment:
There are too many debug logs, make sure by default debug logs are not printed in service. Else this will pollute the production logs a lot

Reply:
debug logs are not printed by default.